pagination/src/main/java/com/cerner/beadledom/pagination/parameters/LimitParameter.java

```diff
@@ -0,0 +1,98 @@
+package com.cerner.beadledom.pagination.parameters;
+
+import com.cerner.beadledom.pagination.BeadledomPaginationModule;
+import com.cerner.beadledom.pagination.parameters.exceptions.InvalidParameterException;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.wordnik.swagger.annotations.ApiModel;
+import com.wordnik.swagger.annotations.ApiModelProperty;
+
+import javax.validation.constraints.Pattern;
+
+/**
+ * Represent the limit parameter used for pagination.
+ *
+ * <p>The validation rules needed for a limit parameter are contained here
+ * so that the same rules may be applied consistently across all paged
+ * resources.
+ *
+ * @author Brian van de Boogaard
+ * @author Will Pruyn
+ * @since 2.9
+ */
+@ApiModel
+public class LimitParameter extends AbstractParameter<Integer> {
+
+  @ApiModelProperty(value = "Total number of items to return in the response.", dataType = "int",
+      allowableValues = "range[0, 100]")
```

Comment:
The swagger doc will be wrong if non-default values are used.

Reply:
This and your comment on the limit check raises an interesting question. Our swagger doc and the validations only mention the acceptable range of values provided by the caller, which isn't currently configurable. The configurable piece is the default value when nothing is provided by the caller. That means the default value could be configured to be outside the range of acceptable values, and everything will function fine.

For example, you can configure the default limit to be `1,000`, and it will succeed and the prev/next links will be correct etc. But if the caller follows the next link and actually _specifies_ `1,000` for the limit it will throw an error due to it being above `100`. 

All this to say: should the minimum limit, maximum limit, and minimum offset be configurable? If yes, the documentation cannot actually mention those limits since the annotations can't include variables. It would also add even more constructors to the pagination module.

Reply:
After some offline discussion it was decided to allow the maximum limit to be configurable.